### PR TITLE
This community NEEDS to adopt a code of conduct. Recent events make t…

### PR DESCRIPTION
**Pull request by jrmithdobbs**
…his glaringly obvious.

### CODE REVIEW

File: CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+By adopting this Code of Conduct, project maintainers commit themselves to
+fairly and consistently applying these principles to every aspect of managing
+this project. Project maintainers who do not follow or enforce the Code of
+Conduct may be permanently removed from the project team.
+
+This Code of Conduct applies both within project spaces and in public spaces
+when an individual is representing the project or its community.
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting a project maintainer at [INSERT EMAIL ADDRESS]. All
```
Comment (jonasschnelli) [inline CONDUCT.md]: `[INSERT EMAIL ADDRESS]` -> use `IRC channel #bitcoin-core-dev` or `https://github.com/bitcoin/bitcoin/issues` or `https://bitcoin.org/en/development`?

Comment (jrmithdobbs) [inline CONDUCT.md]: whoops.

**Comment by jrmithdobbs** [comment]
Let me know if you'd like me to squash.

**Comment by jrmithdobbs** [comment]
I want to be clear here. I am a staunch supporter of free speech and have previously been opposed to such things. My mind has been changed, not just by this project, by watching friends and talented engineers berated, harassed, and generally have their mental health destroyed.

**Comment by jrmithdobbs** [comment]
I'm unsure as to whether the current contact language is appropriate as I agree with said policy that anonymity preservation should be an option when reporting.

Input?

**Comment by bitstein** [comment]
> I am a staunch supporter of free speech

No you aren't.

**Comment by jrmithdobbs** [comment]
@bitstein thank you for illustrating the need so immediately.

**Comment by justusranvier** [comment]
@jrmithdobbs Should your comment be interpreted as stating that @bitstein's comment would be forbidden if this policy is adopted?

in other words, the reason to implement a code of conduct is to allow enforcers of the code to ban anyone who disagrees with them?

**Comment by jrmithdobbs** [comment]
No, just an illustration of the current toxic environment revolving around bitcoin.

It should definitely be interpretted as a complete divestment of any endorsement of bitcointalk.org, however, sorry.

**Comment by jrmithdobbs** [comment]
To be clear, related PRS:

bitcoin-dot-org/bitcoin.org#1167
bitcoin/secp256k1#369
libbitcoin/libbitcoin#355

**Comment by PierreRochard** [comment]
Troll laying bait, as evidenced by https://twitter.com/jrmithdobbs/status/679036167157735425

I'm ok with a code of conduct, I'm not ok with abusive pull requests that violate said code of conduct, like this one.

**Comment by jrmithdobbs** [comment]
I'm not trolling. The comment on twitter is my expectations of the response of this community.

**Comment by jrmithdobbs** [comment]
You should search full-disclosure for mt gox.

**Comment by jrmithdobbs** [comment]
So far, expectations met:

https://github.com/libbitcoin/libbitcoin/pull/355

**Comment by petertodd** [comment]
@jrmithdobbs Would you please provide some specific examples of conduct directly related to Bitcoin Core in the past few months that would be against this proposed code of conduct.

**Comment by unclediazepam** [comment]
FYI PR author is a known troll and bad actor:

https://storify.com/willrad/breaking-linux-user-is-kind-of-an-asshole

**Comment by jrmithdobbs** [comment]
@petertodd nothing within bitcoin core specifically. This is a sincere request to codify the fact that such things should not be tolerated. I do not think it's unreasonable and discussion on phrasing or choosing a different specific wording for such a code of conduct is fine by me. I think the trend towards these types of CoC is a good thing and merely wish it to be extended to a project I care about.

Additionally I'll not be responding to the personal attacks being issued by that puppet account with no commits that seems to have been created by someone in the libbitcoin PR.

**Comment by petertodd** [comment]
If there's no CoC violations in Bitcoin Core, NACK for now. Open to reconsidering in the future if the situation changes.

On 21 December 2015 15:09:41 GMT-08:00, Douglas Huff <notifications@github.com> wrote:
>@petertodd nothing within bitcoin core specifically. This is a sincere
>request to codify the fact that such things should not be tolerated. I
>do not think it's unreasonable and discussion on phrasing or choosing a
>different specific wording for such a code of conduct is fine by me. I
>think the trend towards these types of CoC is a good thing and merely
>wish it to be extended to a project I care about.
>
>Additionally I'll not be responding to the personal attacks being
>issued by that puppet account with no commits that seems to have been
>created by someone in the libbitcoin PR.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/bitcoin/bitcoin/pull/7243#issuecomment-166452125

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jrmithdobbs** [comment]
That's fair but I personally think waiting for the bad things to happen is the wrong approach.

**Comment by Jogyn** [comment]
Please do the responsible thing and fork and create your own community with your own rules. Stop trying to unnecessarily safespace communities. It is counter productive.

**Comment by unclediazepam** [comment]
@Jogyn that would require doing actual work rather than spending 30 minutes to send 4 PRs that disrupt and polarize open source communities who are actually building things.

**Comment by jrmithdobbs** [comment]
It's well documented that I'm an asshole. I want to ensure you that this PR was
made in good faith. I'm trying to do better w/ the asshole thing but things are
hard to change over night. I only opened the libbitcoin one because I didn't
realize @genjix was no longer lead and assumed he would be all for it. I did not
expect it to end in literal racial slurs. I am absolutely appaled by that. libbitcoin/libbitcoin#355

I've recently had some experiences and seen examples of why things like this
are useful. I used to oppose them too. But I've been convinced, oddly enough by
@hintjens and a few others' extremely eloquent writing on the subject combined
with the Clusterfuck of "bro" culture I've encountered since moving to the Bay
Area.

Please don't think I'm trolling. I'm not going to answer the personal attacks,
as I've stated, it's not worth it.

I would honestly like a direct declaration of support of a similar CoC in some
form. Our *industry* (not crypto currency, tech) is hurting itself with strict
adherence to its own idealism and excluding very useful contributors.

The recent glaringly obvious comment was in reference to this insane "Chinese
miner" vs the world nonsense going on with personal attacks flying everywhere.
It's disgusting especially at a time of such huge wins re: criminal prosecution
of fraudsters and acceptance by the SEC for securities issuance.

I know it's not being encouraged by the core devs and I think you know that I
respect the vast majority of you guys even if my tone is harsh and word choice
horrible at times. Again, I know it's not coming from core devs but there are other
projects that encourage similar and I just think a bold, clear, statement to
that effect by the core guys may go a long way to making this community a
better place for everyone.

I knew associating my name with it would steal some of the concept's
credibility. But like others in this community have said. "Anything worth doing
is worth putting your name on." No one else was proposing so I did.

I probably should have been more thorough in uniquely differentiating these
different pull requests and can understand how it could seem to be imposing a
bad impression. That is my bad and totally on me. I sincerely apologize for
that. Do not let my communication skills derail discussion of, what I believe
to be, a very important topic for *any* community but especially this one of as
of late.

(just to get them linked in the others: bitcoin-dot-org/bitcoin.org#1167 bitcoin/secp256k1)

**Comment by jrmithdobbs** [comment]
I will address one attack: I am not an outsider and the intended audience of this specific PR know this.

**Comment by unclediazepam** [comment]
>  I'm trying to do better w/ the asshole thing

So you disrupt 4 open source projects with your CoC PR and brag about it on Twitter?

Try harder.

**Comment by jrmithdobbs** [comment]
I sincerely wish I had screenshotted @unclediazepam's use of "BTFO SJW" since he's wisely deleted it to try and make the above nonsensical.

**Comment by jrmithdobbs** [comment]
Oh wait. I did. https://pbs.twimg.com/media/CWyJsZWUwAAP6gL.jpg

**Comment by evoskuil** [comment]
Re: the libbitcoin PR
> I did not expect it to end in literal racial slurs

"Fuck Off" and "douchetard" are not racial slurs dumbass.

**Comment by dcousens** [comment]
Lock PR?

**Comment by petertodd** [comment]
ACK lock PR

On 21 December 2015 18:06:08 GMT-08:00, Daniel Cousens <notifications@github.com> wrote:
>Lock PR?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/bitcoin/bitcoin/pull/7243#issuecomment-166478837

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jrmithdobbs** [comment]
For the good of everyone involved I'll happily close. This devolved into nonsense I half expected but hoped wouldn't occur. I'm sorry.

**Comment by jrmithdobbs** [comment]
I am still interested in a constructive conversation on this topic and would like advice on how to elicit such constructively.

**Comment by laanwj** [comment]
I've seen some devious, manipulative behavior in the bitcoin-core developer community but none that the basic kindergarten rules outlined here would prevent. *This* kind of behavior usually comes from outside, e.g. "vote" floods from reddit and no, those people won't read or heed a document like this any way - they expect to be banned and create an account just for that.

> I'm ok with a code of conduct, I'm not ok with abusive pull requests that violate said code of conduct, like this one.

Same here. I don't think it would hurt to write down how we, as a developer community, intend to behave toward each other.
But that would have to apply to specific incidents in our community and not a copy/paste code of conduct which misses the buck.

> I am still interested in a constructive conversation on this topic and would like advice on how to elicit such constructively.

Right, after first trying to inject a code of conduct into a project with the intent to create a shitstorm. You may not be violating the letter of your own code of conduct but certainly the spirit. This does kind of remove any motivation to include you in constructive conversation about this subject.

**Pull request closed without merging**